lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_login.rb

```diff
@@ -0,0 +1,59 @@
+require 'googleauth'
+require 'googleauth/stores/file_token_store'
+require "google/apis/people_v1"
+require "fileutils"
+
+module Fastlane
+  module Actions
+    class FirebaseAppDistributionLoginAction < Action
+      OOB_URI = "urn:ietf:wg:oauth:2.0:oob"
+      SCOPE = 'https://www.googleapis.com/auth/cloud-platform'
+      CLIENT_ID = "563584335869-fgrhgmd47bqnekij5i8b5pr03ho849e6.apps.googleusercontent.com"
+      CLIENT_SECRET = "j9iVZfS8kkCEFUPaAeJV0sAi"
+
+      def self.run(params)
+        client_id = Google::Auth::ClientId.new(CLIENT_ID,CLIENT_SECRET)
+        token_store = nil
+        authorizer = Google::Auth::UserAuthorizer.new(client_id, SCOPE, token_store)
+        url = authorizer.get_authorization_url(base_url: OOB_URI)
+
+        UI.message("Please open the following address in your browser:")
+        UI.message(url.to_s)
+        UI.message("")
+        code = UI.input("Enter the resulting code here: ")
+        credentials = authorizer.get_credentials_from_code(code: code, base_url: OOB_URI)
+        token = credentials.refresh_token
+
+        UI.message("Refresh Token: #{token}")
```

Comment:
The `token` variable isn't used anywhere else so consider inlining i.e. `UI.message("Refresh Token: #{credentials.refresh_token}")`.